### PR TITLE
Home screen filled with mock data

### PR DESCRIPTION
**Pull request by lukan98**
Feature demonstration:
https://user-images.githubusercontent.com/53893982/126993708-6990d6af-a6e3-44a6-8fab-f511850b1d30.mp4

### CODE REVIEW

File: MovieApp/DomainLayer/Models/GenreModel.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+struct GenreModel {
+
+    let id: Int
+    let name: String
+
+}
+
+// MARK: RepositoryModel to Model conversion
+
```
Comment (istaudacher) [inline MovieApp/DomainLayer/Models/GenreModel.swift]: remove this empty line

File: MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView+Design.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    func animatedDataReload() {
+        UIView.animate(
+            withDuration: 0.4,
+            animations: {
+                self.movieCollectionView.alpha = 0.5
+                self.movieCollectionView.reloadData()
+                self.movieCollectionView.setContentOffset(.zero, animated: true)
+                self.movieCollectionView.alpha = 1
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView+Design.swift:53]: Hm, you are setting alpha to 0.5 in line 50, which one is the one you want?

File: MovieApp/PresentationLayer/Models/GenreViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
+struct GenreViewModel {
+
+    let id: Int
+    let name: String
+
+}
+
+// MARK: Model to ViewModel conversion
+
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Models/GenreViewModel.swift]: remove this line

**Review by istaudacher — CHANGES_REQUESTED** [review]

File: MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView+Design.swift (hunk 2 above)
Comment (lukan98) [reply MovieApp/PresentationLayer/HomeScreen/CategorisedCollection/CategorisedCollectionView+Design.swift:53]: I wanted a smoother transition for the data reload, but setting the alpha to 0 and then to 1 was a bit jarring so I settled on having alpha go to 0.5 then to 1 so the animation is a bit easier on the eyes.
However, I'm not sure if this is the best way to do this.

**Review by istaudacher — APPROVED** [review]

**Pull request merged**